src/renderer/stores/Workspace.ts

```diff
@@ -59,6 +61,14 @@ export class Workspace {
       this.showSidebar = !this.showSidebar;
       return false;
     });
+    Mousetrap.bind(["command+o", "ctrl+o"], () => {
+      this.openProject();
```

Comment:
A shortcut is a good idea. But we should close the current project so it would terminate the current runner if its running and clear the files. `this.closeProject()`

Reply:
I think a better idea is to put the call to closeProject() within openProject. a la:

```javascript
  openProject() {
    openProjectFolder().then((projectDirectory: string[]) => {
      this.closeProject() // maybe here a check to see if there's a project open.
      const rootPath = projectDirectory[0];
      if (this.validateProject(rootPath)) {
        this.preference.initialize(rootPath);
        this.loadTestFiles();
        this.initializeRunner();
      }
    });
  }
```

This means if you hit open project, and then change your mind and exit the dialog - the original project will remain open. But is there a nice way to check if there's a project open already before calling `closeProject()`? From testing, it doesn't seem to cause any issues if there isn't a project open.

Reply:
I agree. Workspace has a get attribute called `isProjectAvailable` which checks if there is a root path available.